### PR TITLE
Update to LibHac 0.5.0

### PR DESCRIPTION
**Pull request by Thealexbarney**
Many LibHac types will now return or throw result codes the same as Horizon code would.
Small IFileSystem behavior details now match their behavior in Horizon.
Removed open-file tracking in IFileSystem.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/FspSrv/IFileSystem.cs [hunk 1]
```diff
@@ -164,31 +101,15 @@ public long DeleteDirectory(ServiceCtx context)
...
-            catch (UnauthorizedAccessException)
+            catch (HorizonResultException ex)
             {
-                Logger.PrintError(LogClass.ServiceFs, $"Unable to access {name}");
-
-                throw;
+                return ex.ResultValue.Value;
             }
-
+
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/FspSrv/IFileSystem.cs]: remove those empty space

File: Ryujinx.HLE/HOS/Services/FspSrv/IFile.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using Ryujinx.HLE.HOS.Ipc;
 using System;
 using System.Collections.Generic;
+using LibHac;
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/FspSrv/IFile.cs]: Please sort using

File: Ryujinx.HLE/HOS/Services/FspSrv/IFileSystemProxy.cs [hunk 3]
```diff
@@ -241,39 +244,63 @@ public void LoadSaveDataFileSystem(ServiceCtx context)
             SaveDataType    saveDataType = (SaveDataType)context.RequestData.ReadByte();
             SaveInfo        saveInfo     = new SaveInfo(titleId, saveId, saveDataType, userId, saveSpaceId);
             string          savePath     = context.Device.FileSystem.GetGameSavePath(saveInfo, context);
-            LocalFileSystem fileSystem   = new LocalFileSystem(savePath);
 
-            DirectorySaveDataFileSystem saveFileSystem = new DirectorySaveDataFileSystem(fileSystem);
+            try
+            {
+                LocalFileSystem fileSystem = new LocalFileSystem(savePath);
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/FspSrv/IFileSystemProxy.cs]:
```suggestion
                LocalFileSystem             fileSystem     = new LocalFileSystem(savePath);
                DirectorySaveDataFileSystem saveFileSystem = new DirectorySaveDataFileSystem(fileSystem);
```

File: Ryujinx.HLE/HOS/Services/FspSrv/IStorage.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 using Ryujinx.HLE.HOS.Ipc;
 using System.Collections.Generic;
+using LibHac;
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/FspSrv/IStorage.cs]: Please sort using.

**Pull request merged**